src/libostree/ostree-repo-pull.c

```diff
@@ -5204,6 +5207,13 @@ find_remotes_cb (GObject      *obj,
       const CommitMetadata *latest_commit_metadata = NULL;
       g_autofree gchar *latest_commit_timestamp_str = NULL;
 
+      if (override_commit_ids)
+        {
+          g_debug ("%s: Using specified commit ‘%s’ for ref (%s, %s).",
+                   G_STRFUNC, override_commit_ids[i], refs[i]->collection_id, refs[i]->ref_name);
+          continue;
+        }
```

Comment:
It’s probably worth adding a comment above this explaining that if this path is taken, `ref_to_latest_commit` is no longer valid.